### PR TITLE
add missing lock to crypter GetKeys()

### PR DESCRIPTION
**Pull request by benma**
Issue: #10905 

First commit makes GetKeys() return the result instead of writing to a reference to remove some useless lines.

### CODE REVIEW

File: src/wallet/crypter.h [hunk 1]
```diff
@@ -162,28 +162,26 @@ class CCryptoKeyStore : public CBasicKeyStore
     {
         {
```
Comment (benma) [inline src/wallet/crypter.h]: Could this scope level be removed alongside with the `return false` in the end, or am I missing something about how `LOCK` works?

Comment (corebob) [inline src/wallet/crypter.h]: @benma It certainly looks like it. The lock goes out of scope either way, and that last return is unreachable as it stands now

**Comment by jonasschnelli** [comment]
I guess the `setAddress` pass by reference was a humble try to keep the memory control on the caller side and have it properly cleaned. But seems not to be done well.

The lock in `GetKeys` is definitively missing. Also, the `IsCrypted()` method is sometimes called without the `cs_KeyStore` so, `fUseCrypto` should be turned into an atomic.

It was never a problem because – at the moment – GetKeys get only called when holding cs_main/cs_wallet.

utACK fe09b0197c20dc3c0a614c1a94dac708ef206743

**Comment by TheBlueMatt** [comment]
utACK fe09b0197c20dc3c0a614c1a94dac708ef206743. As @jonasschnelli points out we need to also fix fUseCrypto to either be atomic or under the lock.

**Comment by jonasschnelli** [comment]
Tested ACK fe09b0197c20dc3c0a614c1a94dac708ef206743
Ready for merge. Another pair of eyes would be good.

**Pull request merged**